nemo/core/classes/modelPT.py

```diff
@@ -455,7 +455,7 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
         if 'sched' in optim_config and self._trainer is not None:
             if not isinstance(self._trainer.accumulate_grad_batches, int):
                 raise ValueError("We do not currently support gradient acculumation that is not an integer.")
-            if self._trainer.max_steps is None:
+            if self._trainer.max_steps is None or self.trainer.max_steps <= 0:
```

Comment:
Can you comment on why we need <=0: Like # Lightning maps max_steps = None to -1

Reply:
Well my idea was 0 max steps is odd no ? I can revert it to < 0 though

Reply:
Reverted to < 0